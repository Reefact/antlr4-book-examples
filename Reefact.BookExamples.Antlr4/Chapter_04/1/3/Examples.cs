#region Usings declarations

using System.Text;

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Reefact.BookExamples.Antlr4.Chapter_04._1._2;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_04._1._3 {

    [UseReporter(typeof(CustomReporter))]
    public class Examples {

        [Fact]
        public void lisp_style_tree_example() {
            // Setup
            StringBuilder example = new();
            example.AppendLine("(1+2");
            example.AppendLine("3");
            AntlrInputStream inputStream = AntlrInputStreamReader.Read(example);
            // Exercise
            GRun   grun          = GRun.Read(inputStream);
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void mermaid_style_tree_example() {
            // Setup
            StringBuilder example = new();
            example.AppendLine("(1+2");
            example.AppendLine("34*69");
            AntlrInputStream inputStream = AntlrInputStreamReader.Read(example);
            // Exercise
            GRun   grun             = GRun.Read(inputStream);
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}