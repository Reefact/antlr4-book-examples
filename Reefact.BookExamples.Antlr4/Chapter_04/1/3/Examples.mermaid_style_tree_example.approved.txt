graph TD
	1["prog"] --> 2["stat"]
	2 --> 3["expr"]
	3 --> 4["("]
	3 --> 5["expr"]
	5 --> 6["expr"]
	6 --> 7["1"]
	5 --> 8["+"]
	5 --> 9["expr"]
	9 --> 10["2"]
	3 --> 11["#0060;missing ')'#0062;"]
	2 --> 12["#0092;r#0092;n"]
	1 --> 13["stat"]
	13 --> 14["expr"]
	14 --> 15["expr"]
	15 --> 16["34"]
	14 --> 17["*"]
	14 --> 18["expr"]
	18 --> 19["69"]
	13 --> 20["#0092;r#0092;n"]
